src/Routes/Shopping-Cart/ShoppingCart.jsx

```diff
@@ -0,0 +1,163 @@
+      } finally {
+        setIsLoading(false)
+      }
+    }
+
+  const handleSearch = (query) => {
+    const searchTerms = query.split(',').map(term => term.trim().toLowerCase())
+    if(query.length > 0){
+       const searchedProduct = [...product].filter((item) => searchTerms.some(term => item.name.toLowerCase().includes(term)))
+       setProduct(searchedProduct)
+    }else {
+     setProduct(testingArray);
+    }
+  }
+
+  const handleSortAZ = () => {
+    const sortedProducts = [...product].sort((a, b) => a.name.localeCompare(b.name));
+    setProduct(sortedProducts)
+  }
+
+    const handleSortZA = () => {
+    const sortedProducts = [...product].sort((a, b) => b.name.localeCompare(a.name));
+    setProduct(sortedProducts)
+  }
+
+  const handleSortByPriceDescAndAZ = () => {
+  const sortedProducts = [...product].sort((a, b) => {
+    const numericPriceA = parseFloat(a.price.replace('$', ''))
+    const numericPriceB = parseFloat(b.price.replace('$', ''))
+    if (numericPriceB !== numericPriceA) {
+      return numericPriceB - numericPriceA;
+    }
+    return a.name.localeCompare(b.name);
+  });
+  setProduct(sortedProducts)
+}
+
+const handleSortByPriceAscAndAZ = () => {
+  const sortedProducts = [...product].sort((a, b) => {
+    const numericPriceA = parseFloat(a.price.replace('$', ''))
+    const numericPriceB = parseFloat(b.price.replace('$', ''))
+    if (numericPriceA !== numericPriceB) {
+      return numericPriceA - numericPriceB;
+    }
+    return a.name.localeCompare(b.name);
+  });
+  setProduct(sortedProducts);
+}
+
+ const incrementQuantity = (index) => {
+    const updatedProducts = [...product];
+    updatedProducts[index].quantity += 1;
+    setProduct(updatedProducts);
+  };
+
+ const decrementQuantity = (index) => {
+    const updatedProducts = [...product];
+    if (updatedProducts[index].quantity > 1) {
+      updatedProducts[index].quantity -= 1;
+    }
+    setProduct(updatedProducts);
+  };
+
+const calculateTotalPrice = () => {
+  const totalPrice = product.reduce((accumulator, product) => {
+    const numericPrice = parseFloat(product.price.replace('$', ''))
+    return accumulator + numericPrice * product.quantity;
+  }, 0)
+  return totalPrice
+}
+
+const totalPurchasePrice = calculateTotalPrice()
+
+  if (isLoading || !product) return <Loader />
+
+ return(
+  <section className='mt-8 m-auto md:max-w-[800px] md:mt-14 md:pb-5 md:rounded-3xl md:border md:border-solid md:border-slate-50 flex flex-col items-center '>
+   < SearchInput onSearch={handleSearch} />
+   <div className='flex gap-5 mt-4 items-center justify-center flex-wrap px-3'>
+    <h2 className='w-full text-center text-base md:text-xl'>Sort by:</h2>
+    < SortButton text='Higher price' onClick={handleSortByPriceDescAndAZ} />
+    < SortButton text='Lower price' onClick={handleSortByPriceAscAndAZ}  />
+    < SortButton text='A-Z' onClick={handleSortAZ} />
+    < SortButton text='Z-A' onClick={handleSortZA}  />
```

Comment:
Ree un detalle, pero no se suele dejar un espacio entre el `<` y el comienzo del nombre del componente 
```suggestion
   <SearchInput onSearch={handleSearch} />
   <div className='flex gap-5 mt-4 items-center justify-center flex-wrap px-3'>
    <h2 className='w-full text-center text-base md:text-xl'>Sort by:</h2>
    <SortButton text='Higher price' onClick={handleSortByPriceDescAndAZ} />
    <SortButton text='Lower price' onClick={handleSortByPriceAscAndAZ}  />
    <SortButton text='A-Z' onClick={handleSortAZ} />
    <SortButton text='Z-A' onClick={handleSortZA}  />
```